### ISSUE TITLE
(macOS) Muting has no effect: There are still sounds

### ISSUE THREAD

**Issue by JuicyJammy** [minetest/minetest#14100]
### Minetest version

```true
minetest 5.8.0
```

### Irrlicht device

X11

### Operating system and version

macOS Ventura

### CPU model

Apple M1

### Summary

Even after pressing the Mute key or adjusting the sound to 0, the game is still making noises.

### Steps to reproduce

1. Create a world in whatever game
2. Click M for muting
3. It says Muted
4. Sounds were still made

**Comment by Desour** [comment]
Can't reproduce.

Note that mute doesn't work for main menu music in 5.8.0 (was fixed in 5.9.0-dev).

Does lowering the sound volume with the minetest setting have any effect?

Did this already happen for you in 5.7?

Which openal implementation do you have installed?

**Comment by Warr1024** [comment]
I can confirm.

This has been my experience with every version of MT on mac that I have used over the past several years (going back to at least 5.6 or 5.5, probably earlier).  Neither mute toggle nor minetest's volume control have any effect.  In my case, it's mainly in-game sounds that are affected, not just the main menu.

I have been using the prebuilt version installed by homebrew.

My current test case:

```
Minetest 5.8.0 (OSX)
Using Irrlicht 1.9.0mt13
Using LuaJIT 2.1.1700008891
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="minetest.app/Contents/Resources"
STATIC_LOCALEDIR="minetest.app/Contents/Resources/locale"
```

How do I find out what OpenAL it was built with?  I assume that if it's homebrew, it won't necessarily depend on a system-wide implementation and may install its own as a dependency, or statically compiled in or something.

**Comment by Warr1024** [comment]
N.B. I have only ever seen this on Mac, not on any other platform (at least Linux and Android) so it may help to update the issue title to say MacOSX unless somebody can confirm it affects others.

I have only ever tested on an Intel Macbook, so given that the original report is on Apple M1, this seems to be related to OSX but not necessarily to chip architecture.

**Comment by Desour** [comment]
It looks like we don't install any library for openal in the macos builds, see `install_macos_deps` in `util/common.sh`.
I guess this means we're using Apple's "Core Audio": https://developer.apple.com/library/archive/documentation/MusicAudio/Conceptual/CoreAudioOverview/CoreAudioEssentials/CoreAudioEssentials.html

OpenAL-soft (the lib that most linux distros for example use by default) seems to also be available for macos, e.g. [via homebrew](https://formulae.brew.sh/formula/openal-soft).
Try using that. (Idk if the cmakelists need to be updated, or if you even need to rebuild.)

**Comment by sfence** [comment]
OpenAL looks to be part of macOS. There is a framework `OpenAL.framework` (at least from macOS 10.15).
So there is no need to install.

**Comment by Desour** [comment]
> OpenAL looks to be part of macOS. There is a framework `OpenAL.framework` (at least from macOS 10.15). So there is no need to install.

But you *can* install it. And if macos's openal implementation has issues, we should recommend users to install openal-soft.
Also note that macos's openal implementation deprecated: https://developer.apple.com/documentation/macos-release-notes/macos-catalina-10_15-release-notes#Audio

**Comment by sfence** [comment]
@Desour or fix it with a workaround: #15128

**Comment by sfence** [comment]
@JuicyJammy Can you test this build https://github.com/minetest/minetest/actions/runs/10748718595/artifacts/1903970625 ?
